### ISSUE TITLE
Commit path can add unstaged issues.jsonl normalization churn after index was verified clean

### ISSUE THREAD

**Issue by Dicklesworthstone** [Dicklesworthstone/beads_rust#337]
## Summary

In a live multi-agent checkout, a source-only commit ended up including `.beads/issues.jsonl` even though the staged index was verified to contain only three non-Beads files immediately before `git commit`.

This is related to #326, but it is a distinct safety failure. #326 covers diagnostic/status commands dirtying JSONL. This report is about the commit path or hook path mutating/staging tracker export churn after an agent has already checked the index and before Git creates the commit.

## Environment

```text
workspace: /Users/jemanuel/projects/eidetic_engine_cli
repo branch: main
br --version: br 0.1.13
gh account: Dicklesworthstone
observed: 2026-06-07
commit produced: e49b7e94
git parent: 682d9a02
```

The workspace is a crowded agent checkout. Other source files were dirty, and `.beads/issues.jsonl` had been dirty earlier in the session, but the staged index was explicitly narrowed and checked before committing.

## Exact command sequence

The intended source-only slice changed only these files:

```text
docs/agent-ux/dueling-wizards-rejected-ideas-register.md
tests/contracts/dueling_wizards_rejected_ideas_register.rs
tests/fixtures/contracts/dueling_wizards_rejected_ideas_register.json
```

Immediately before committing, I staged only those paths and verified the index:

```bash
git add -- \
  docs/agent-ux/dueling-wizards-rejected-ideas-register.md \
  tests/contracts/dueling_wizards_rejected_ideas_register.rs \
  tests/fixtures/contracts/dueling_wizards_rejected_ideas_register.json

git diff --cached --name-only
```

Observed staged names:

```text
docs/agent-ux/dueling-wizards-rejected-ideas-register.md
tests/contracts/dueling_wizards_rejected_ideas_register.rs
tests/fixtures/contracts/dueling_wizards_rejected_ideas_register.json
```

I then checked the staged diff:

```bash
git diff --cached --stat
git diff --cached --check
```

Observed staged stat:

```text
 .../dueling-wizards-rejected-ideas-register.md     |  15 ++
 .../dueling_wizards_rejected_ideas_register.rs     | 178 +++++++++++++++++++++
 .../dueling_wizards_rejected_ideas_register.json   | 134 ++++++++++++++++
 3 files changed, 327 insertions(+)
```

`git diff --cached --check` was clean.

Then I committed:

```bash
AGENT_NAME=VioletCove git commit -m "test: add rejected ideas decision coverage matrix"
```

The commit succeeded, but Git reported:

```text
[main e49b7e94] test: add rejected ideas decision coverage matrix
 4 files changed, 3917 insertions(+), 3590 deletions(-)
```

## Actual commit contents

```bash
git show --format= --name-only HEAD
```

Observed:

```text
.beads/issues.jsonl
docs/agent-ux/dueling-wizards-rejected-ideas-register.md
tests/contracts/dueling_wizards_rejected_ideas_register.rs
tests/fixtures/contracts/dueling_wizards_rejected_ideas_register.json
```

The `.beads/issues.jsonl` diff is a full-file normalization/export shape:

```bash
git diff --numstat HEAD^ HEAD -- .beads/issues.jsonl
```

```text
3590    3590    .beads/issues.jsonl
```

Hashes and line counts:

```text
HEAD^ .beads/issues.jsonl sha256: 679f4bd67a003551a1c2f8ec6ce6800cc86eeb1786b7d1d9265479a997213135
HEAD  .beads/issues.jsonl sha256: 0369bdf5b3609c6eb7ceee065d5c6390edf5a03adf32ba5dce8cb6e65c13b733
HEAD^ line count: 3590
HEAD  line count: 3590
```

Example record-level churn for `bd-00b9`:

Before:

```json
{
  "issue_id": "bd-00b9",
  "depends_on_id": "bd-2k2q",
  "type": "blocks",
  "created_at": "2026-06-07T18:55:57Z",
  "created_by": "import",
  "metadata": "{}",
  "thread_id": ""
}
```

After:

```json
{
  "issue_id": "bd-00b9",
  "depends_on_id": "bd-2k2q",
  "type": "blocks",
  "created_at": "2026-06-07T20:09:22Z",
  "created_by": "import"
}
```

So the commit path did not merely pick up one explicit tracker edit. It committed a broad import/export normalization pass with dependency timestamps and omitted empty/default fields.

## Current doctor/status context

After the commit, Beads reports DB and JSONL in sync but still has a merge-artifact warning:

```json
{
  "ok": true,
  "warnings": [
    {
      "name": "jsonl.merge_artifacts",
      "status": "warn",
      "message": "Merge artifacts detected in .beads/",
      "details": { "files": ["beads.base.jsonl"] }
    }
  ],
  "sync": {
    "name": "sync.metadata",
    "status": "ok",
    "message": "Database and JSONL are in sync",
    "details": {
      "dirty_issues": 0,
      "last_import": "2026-06-07T16:22:43.197427-04:00",
      "last_export": "2026-06-07T20:22:25.097763+00:00",
      "jsonl_hash": "0369bdf5b3609c6e"
    }
  }
}
```

## Expected behavior

A Beads hook or commit integration must not add `.beads/issues.jsonl` to a commit after the caller has already staged and verified a narrower source-only index.

Safer acceptable behaviors:

1. If Beads needs to export/normalize JSONL during commit, fail the commit and instruct the user/agent to make a separate tracker-export commit.
2. If Beads modifies `.beads/issues.jsonl`, leave it unstaged and fail or warn; do not silently include it in the current source commit.
3. If auto-staging is intentional, it must be opt-in and robot-visible before commit, not discovered only from the final commit summary.
4. Hooks should respect the pre-existing staged set. A source-only staged set should remain source-only.

## Why this matters

In shared agent checkouts, agents routinely isolate source slices from tracker export churn. A hook that can add `.beads/issues.jsonl` after `git diff --cached --name-only` makes that isolation unreliable.

Practical damage:

- Source commits accidentally include thousands of tracker-line changes.
- Agents cannot trust pre-commit staged checks.
- Tracker normalization churn gets mixed with unrelated test/doc changes.
- It becomes unsafe to push without manual audit, especially when `.beads/beads.base.jsonl` warnings are present.
- Existing no-delete/no-reset policies make history repair deliberately difficult, so the tool must prevent the bad commit rather than rely on agents rewriting it later.

## Suggested regression test

Create a fixture repo with:

- tracked `.beads/issues.jsonl` containing stale-but-parseable export content,
- a dirty or normalizable Beads DB state,
- an unrelated source file change staged by explicit path,
- `.beads/issues.jsonl` not staged.

Run the same commit hook path and assert one of:

- commit fails with a clear message that Beads export would change JSONL, or
- commit succeeds and contains exactly the originally staged paths.

It should never succeed with `.beads/issues.jsonl` added to the commit unless the user explicitly staged that path or passed an explicit opt-in flag.

## Related issues

- #326: diagnostic/status commands can rewrite `issues.jsonl` with full dependency metadata churn.
- #328: doctor merge-artifact warning should classify `beads.base.jsonl` and provide safe remediation.
- #330: `doctor.ok` can be true while operational tracker authority is ambiguous or blocked.
- #334: automation gate for shared tracker authority and claimability.

**Comment by Dicklesworthstone** [comment]
Follow-up context: #326 now has a fresh read/list-path reproduction from the same workspace showing how the tracker export can become dirty again before a commit even enters the picture.

After the accidental source commit described here, a read-oriented orchestration command:

```bash
ee swarm brief --workspace . --json
```

used the Beads ready/list path (`br ready --json`) and left `.beads/issues.jsonl` dirty again:

```text
3590 insertions / 3590 deletions
HEAD hash:     0369bdf5b3609c6eb7ceee065d5c6390edf5a03adf32ba5dce8cb6e65c13b733
worktree hash: c8e8dd400689ad59c1b81b518ae1a6ba3ea2b5590d2bbd862e64575b43cf1468
line count:    3590 in both versions
```

The shape flipped defaults back in (`source_repo`, `compaction_level`, `original_size`, dependency `metadata: "{}"`, dependency `thread_id: ""`). `br sync --status --json --no-auto-import --no-auto-flush` then reported `dirty_count: 0`, `jsonl_newer: false`, `db_newer: false`, and blessed the dirty worktree hash as `jsonl_content_hash`.

That strengthens the likely chain for this issue: read/list/status surfaces can create or bless full-file JSONL normalization churn, and then the commit/hook path can silently mix that churn into a source-only commit after the staged set was verified. The fix here still needs to preserve index integrity, but #326 likely needs to stop the upstream read-path mutation/oscillation as well.

**Comment by Dicklesworthstone** [comment]
Resolved as a consequence of #326. The churn that was getting staged came from the read-path `issues.jsonl` export, which is now suppressed (read/status commands no longer export — see #326). `br` itself never stages or commits: `sync` is statically guarded against shelling out to git (`validation/mod.rs:1046-1070` `sync_safety_no_git_commands_in_sync_module()` asserts no `Command::new("git")` in the sync module, and the CLI documents "br sync NEVER executes git commands"). With the read-path export gone, there is no normalization churn for a commit to pick up.
